models/clan.go

```diff
@@ -190,14 +196,42 @@ func (c *Clan) IndexClanIntoElasticSearch() error {
 	return nil
 }
 
+// NewClanWithNamePrefixes returns a new extended Clan object with name  prefixes
+func (c *Clan) NewClanWithNamePrefixes() *ClanWithNamePrefixes {
+	minPrefixLength := 4 // how to bring the app Viper config here?
```

Comment:
The problem is that we update MongoDB through insert/update hooks fired by our ORM/wrapper for Postgres. There is no trivial modification for those parts to receive viper and pass to this function =(

Reply:
It could be done (I just looked at models.Clan code), but it would take some effort. You can leave a TODO here about this and we can change it later. Ok?